tests/extra/job_management/test_thread_worker.py

```diff
@@ -0,0 +1,204 @@
+    def enqueue_and_wait(self, worker, work_item, timeout=0.2):
+        worker.work_queue.put(work_item)
+        worker.start()
+        sleep(timeout)
+        worker.stop_event.set()
+        worker.join(timeout=1)
+
+    def test_run_jobs_collects_worker_results(self, tmp_path, job_manager, sleep_mock):
+        df = pd.DataFrame({"year": [2018], "geometry": ["POINT (1 2)"]})
+        job_db_path = tmp_path / "jobs.csv"
+        job_db = CsvJobDatabase(job_db_path).initialize_from_df(df)
+
+        captured_results = []
+        original_process_result_queue = job_manager._process_result_queue
+
+        def spy_process_result_queue(stats=None):
+            try:
+                work_result = job_manager._result_queue.get_nowait()
+                captured_results.append(work_result)
+            except queue.Empty:
+                pass
+            original_process_result_queue(stats)
+
+        job_manager._process_result_queue = spy_process_result_queue
+
+        run_stats = job_manager.run_jobs(job_db=job_db, start_job=self._create_year_job)
+
+        expected = (
+            _JobManagerWorkerThread.WORK_TYPE_START_JOB,
+            ("job-2018", True, "queued")
+        )
+        assert any(result == expected for result in captured_results), (
+            f"Expected to see {expected!r} among worker results {captured_results!r}"
+        )
+
+    def test_worker_thread_lifecycle(self, fresh_worker):
+        fresh_worker.polling_time = 0.1  
+        fresh_worker.start()
+        assert fresh_worker.is_alive()
+        fresh_worker.stop_event.set()
+        fresh_worker.join(timeout=1)
+        assert not fresh_worker.is_alive()
+
+    def test_start_job_success(self, fresh_worker, requests_mock):
+        backend_url = "https://foo.test"
+        job_id = "job-123"
+
+        requests_mock.get(backend_url, json={"api_version": "1.1.0"})
+        requests_mock.post(
+            f"{backend_url}/jobs",
+            json={"job_id": job_id, "status": "created"},
+            status_code=201,
+            headers={"openeo-identifier": job_id}
+        )
+        requests_mock.post(
+            f"{backend_url}/jobs/{job_id}/results",
+            json={"job_id": job_id, "status": "finished"},
+            status_code=202
+        )
+        requests_mock.get(
+            f"{backend_url}/jobs/{job_id}",
+            json={"id": job_id, "status": "finished"}
+        )
+        
+        self.enqueue_and_wait(fresh_worker, (fresh_worker.WORK_TYPE_START_JOB, (backend_url, "bearer_token_123", job_id)))
+        
+        assert fresh_worker.result_queue.qsize() == 1
+        work_type, (jid, success, status) = fresh_worker.result_queue.get()
+        assert success is True
+        assert status == "finished"
+
+    def test_start_job_failure(self, fresh_worker, monkeypatch):
+        def mock_connect(*args, **kwargs):
+            raise ConnectionError("Backend unreachable")
+        monkeypatch.setattr(openeo, "connect", mock_connect)
```

Comment:
I would mock the http response here like in the other tests instead of monkeypaching openeo.connect